### DISCUSSION TITLE
The difference between the chia consensus algorithm and chia green paper

### DISCUSSION THREAD

**Discussion by yingjio** [Chia-Network/chia-blockchain#12344] [Q&A] [answered]
To whom it may concern,

I am a graduate student from National Cheng Kung University (NCKU), and I have study about chia network recently, and I found that many different between "chia consensus algorithm" and "chia green paper", including the challenge of the Proof of Space and the difficulty adjustment. I wonder which one should I be based on and I would appreciate that if you can solve my confusions.

Many thanks,
Jason

**Comment by mariano54** [comment, marked as answer]
You should be based on the code in the python repo. Also, if you want to learn more about the algorithm, you should read this:
https://docs.chia.net/docs/03consensus/consensus_intro

**Reply by yingjio** [reply]
Thanks a lot !!!
